lib/Transforms/Utils/SimplifyCFG.cpp

```diff
@@ -1258,6 +1258,13 @@ static bool HoistThenElseCodeToIf(BranchInst *BI,
     if (isa<TerminatorInst>(I1))
       goto HoistTerminator;
 
+    // Disallow hoisting of setjmp.  Although hoisting the setjmp technically
+    // produces valid IR, it seems hard to generate appropariate machine code
+    // from this IR, e.g., for X86.
+    if (IntrinsicInst *II = dyn_cast<IntrinsicInst>(I1))
```

Comment:
We should potentially leave an open issue with this as a reminder to return to it later.

Reply:
Sure, feel free to create it.  I've talked to Charles about this bug, and we have ideas for a more precise fix, but the implementation would be substantially more elaborate.  Our current thinking is to save the project of implementing the precise fix for another student.